Добавил компоненты Button и Spinner

GarnagaA opened the pull request:
Добавил компонент Spinner
Добавил компонент Button
issue: нужно импортировать заготовленные цвета из src\scss\colors.scss

Commit:
Добавил компонент Spinner
Добавил компонент Button
issue: импортировать заготовленные цвета

snaxjs commented on src/components/Button/index.scss:
Стили написаны неправильно, смотри примеры стилей в других компонентах

snaxjs commented on src/components/Button/index.tsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <button
+      type="button"
+      className={isLoading? 'storybook-button__spinner' : 'storybook-button'}
```
Неправильно указал стили, читай БЭМ, как указать модификатор.

className={`storybook-button${props.isLoading ? ' storybook-button_loading' : ''}`}

snaxjs commented on src/components/Spinner/index.tsx:
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import {FC} from 'react'
+import './index.scss'
```
Зачем тут импорт стилей ? Это такой же компонент как и все, импортируй в главный файл стилей в папке components

snaxjs commented on src/components/Spinner/index.tsx:
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import {FC} from 'react'
+import './index.scss'
+interface SpinnerProps {
```
interface SpinnerProps { 
  styles?: CSSProperties;
}

snaxjs commented on src/components/Spinner/index.tsx:
```diff
@@ -0,0 +1,20 @@
+    height?: number;
+    margin?: number;
+}
+const Spinner: FC<SpinnerProps> = ({width, height, margin}) => {
```
const Spinner = (props: SpinnerProps}) =>

snaxjs commented on src/components/Spinner/index.tsx:
```diff
@@ -0,0 +1,20 @@
+    margin?: number;
+}
+const Spinner: FC<SpinnerProps> = ({width, height, margin}) => {
+
```
Что за отступы ? Настрой prettier в настройках IDE, иначе будет засчитывать изменения при разработке, гит учитывает это, уже 3 раз говорю тебе, prettier выбрать из папки node_modules ЭТОГО проекта, а не у себя где-то там

snaxjs commented on src/components/Button/Button.stories.tsx:
```diff
@@ -0,0 +1,29 @@
+      />
+  ),
+};
+export const Primary_WithSpinner: Story = {
```
Внимательно почитай документацию, там можно можно указывать пропсы, нужно удалить этот Primary_WithSpinner и сделать это все в одном Primary

snaxjs commented on src/components/Button/index.tsx:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import './index.scss';
```
Убрать импорт

snaxjs commented on src/components/Spinner/index.scss:
```diff
@@ -0,0 +1,38 @@
+.lds-ring {
```
компонент называется spinner, класс должен называться spinner, а не lds-ring

snaxjs commented on src/components/Spinner/index.scss:
```diff
@@ -0,0 +1,38 @@
+}
+
+
+@keyframes lds-ring {
```
spinner-animation, а не lds-ring

snaxjs commented on src/components/Button/index.scss:
```diff
@@ -0,0 +1,42 @@
+  width: 179px;
+}
+
+.storybook-button, .storybook-button__spinner {
```
Стили написаны неправильно, смотри примеры стилей в других компонентах

snaxjs commented on src/components/Button/index.tsx:
```diff
@@ -0,0 +1,28 @@
+}
+
+
+export const Button = ({
```
Компоненты экспортируются по умолчанию, а не так, смотри как экспортируются другие компоненты

Commit:
Исправил стили компонента Button согласно БЭМ

Commit:
issue: Сделать 2 состояния кнопки в одной истории

Commit:
2 состояния кнопки объединил в одну историю

snaxjs commented on src/components/Button/Button.stories.tsx:
```diff
@@ -1,29 +1,53 @@
-const meta: Meta<typeof Button> = {
-  component: Button,
-};
+// const meta: Meta<typeof Button> = {
```
Удали комментарий, не должно быть в проекте такого мертвого кода

snaxjs commented on src/components/Button/Button.stories.tsx:
```diff
@@ -1,29 +1,53 @@
\ No newline at end of file
+};
+
+// export const Primary_WithSpinner: Story = {
```
Удали комментарий, не должно быть в проекте такого мертвого кода

snaxjs commented on src/components/Button/index.scss:
```diff
@@ -1,18 +1,14 @@
+@import "../../scss/colors";
```
Импорт тут не нужен, цвета импортируются в другом месте и они уже импортированы

snaxjs commented on src/components/Button/index.tsx:
```diff
@@ -1,28 +1,28 @@
+import React from "react";
 import Spinner from "../Spinner";
 
 interface ButtonProps {
```
Название интерфейсов начинается с буквы I (Interface), т.е. IButtonProps

snaxjs replied:
Во всех компонентах есть пропс classNames?: string[];

snaxjs commented on src/components/Button/index.tsx:
```diff
@@ -1,28 +1,28 @@
-    > {isLoading && <Spinner width={18} height={18} margin={18}/>}
-      {label}
+      disabled={props.disabled}
+      className={`storybook-button${
```
className={classNames(`storybook-button${
        props.isLoading ? " storybook-button__spinner" : ""
      }`, props.classNames)}

classNames импортируется отсюда https://github.com/snaxjs/rick-and-morty/blob/main/src/utils/ClassNames.ts

GarnagaA replied:
Для каких кейсов этот хелпер classNames ?

snaxjs replied:
Для прокидывания классов как [микс](https://ru.bem.info/methodology/quick-start/#%D0%9C%D0%B8%D0%BA%D1%81)

snaxjs commented on src/components/CharCard/index.scss:
```diff
@@ -14,7 +14,7 @@
     display: flex;
     align-items: center;
-    justify-content: center;
+  justify-content: center;
```
Невнимательно работаешь с кодом или prettier так и не подключил, что за конченный отступ тут ? Причем даже не в твоём компоненте

snaxjs replied:
Должно хватить

snaxjs commented on src/components/Spinner/index.tsx:
```diff
@@ -1,20 +1,16 @@
-import {FC} from 'react'
-import './index.scss'
+import React, { CSSProperties } from "react";
 interface SpinnerProps {
```
ISpinnerProps

snaxjs replied:
Добавь пропс classNames?: string[];

snaxjs commented on src/components/Spinner/index.tsx:
```diff
@@ -1,20 +1,16 @@
\ No newline at end of file
+const Spinner = (props: SpinnerProps) => {
+  return (
+    <div className="spinner" style={props.styles}>
```
className={classNames(storybook-button${ props.isLoading ? " storybook-button__spinner" : "" }, props.classNames)}

classNames импортируется отсюда https://github.com/snaxjs/rick-and-morty/blob/main/src/utils/ClassNames.ts

Commit:
Добавил хелпер classNames;
Добавил classNames: string[];  в пропсы;
Исправил ошибки с написанием названий компонентов без заглавной I ---> ISpinnerProps, IButtonProps  (interface);
Добавил формат scss в отслеживаемые форматы prettier;
